packages/devtools_app/lib/src/network/http_request_data.dart

```diff
@@ -0,0 +1,246 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'http.dart';
+
+/// Contains all state relevant to completed and in-progress HTTP requests.
+class HttpRequests {
+  HttpRequests({this.requests, this.outstanding}) {
+    requests ??= [];
+    outstanding ??= {};
+  }
+
+  void clear() {
+    requests.clear();
+    outstanding.clear();
+  }
+
+  List<HttpRequestData> requests;
+
+  /// A mapping of timeline IDs to instances of HttpRequestData which are
+  /// currently in-progress.
+  Map<String, HttpRequestData> outstanding;
+}
+
+/// Used to represent an instant event emitted during an HTTP request.
+class HttpInstantEvent {
+  HttpInstantEvent._(this._rawEvent) : name = _rawEvent['name'];
+
+  final Map _rawEvent;
+  final String name;
+
+  /// The amount of time since the last instant event completed.
+  double get timeDiffMs => _timeDiffMs;
+  
+  // This is set from within HttpRequestData.
+  double _timeDiffMs;
+}
+
+/// An abstraction of an HTTP request made through dart:io.
+class HttpRequestData {
```

Comment:
order class members according to https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#order-other-class-members-in-a-way-that-makes-sense